pkg/schema/type.go

```diff
@@ -264,10 +294,14 @@ func (a *ArrayType) AssignTypeTo(typeable yamlmeta.Typeable) (chk yamlmeta.TypeC
 	}
 	return
 }
-func (a ArrayItemType) AssignTypeTo(typeable yamlmeta.Typeable) (chk yamlmeta.TypeCheck) {
+func (a *ArrayItemType) AssignTypeTo(typeable yamlmeta.Typeable) (chk yamlmeta.TypeCheck) {
 	arrayItem, ok := typeable.(*yamlmeta.ArrayItem)
 	if !ok {
-		chk.Violations = []string{fmt.Sprintf("Expected node at %s to be a %T, but was a %T", typeable.GetPosition().AsCompactString(), &yamlmeta.ArrayItem{}, typeable)}
+		chk.Violations = append(chk.Violations,
```

Comment:
stylistic nit: would be nice to keep things on one line as long as possible, and then break to second line if it really runs too long.

```
chk.Violations = append(chk.Violations, NewTypeError(
  typeToString(typeable), typeToString(a), typeable.GetPosition(), a.Position))
```

Reply:
Agreed. At times, splitting lines by arguments can significantly improve readability, but this doesn't strike me as such a case.